.github/workflows/dluhc-build-and-publish.yml

```diff
@@ -0,0 +1,190 @@
+
+  setup-env-vars:
+    needs: lint-and-test
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+    steps:
+      - id: set-vars
+        run: |
+          echo "::set-output name=LOWER_CASE_BRANCH::$(
+            echo ${GITHUB_REF#refs/heads/} |
+            tr "[:upper:]" "[:lower:]"
+          )"
+      - name: echo-vars
+        id: echo-vars
+        run: |
+          echo ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+          echo $VERSION
+
+  build-and-push-images:
+    needs: [lint-and-test, setup-env-vars]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [designer, runner]
+    name: build-and-push-${{matrix.app}}-image
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      - name: Docker compose pull
+        run: docker-compose pull
+      - name: Docker layer caching DAYOFYEAR key
+        # DAYOFYEAR key is used to rotate docker-layer-caching daily to avoid the layers dangle issues
+        # see https://github.com/satackey/action-docker-layer-caching/issues/55
+        # if you change this, please remember to apply the same changes to branch--lint-unit-and-smoke-test.yml
+        # so main branch cache can be shared with branches
+        run: echo "DAYOFYEAR=$(date +%j)" >> $GITHUB_ENV
+      - name: Create .env for ${{ matrix.app }} workspace
+        run: |
+          touch ./${{ matrix.app }}/.env
+          echo LAST_TAG_GH=$VERSION >> ./${{ matrix.app }}/.env
+          echo LAST_COMMIT_GH=${{ github.sha }} >> ./${{ matrix.app }}/.env
+          cat ./${{ matrix.app }}/.env
+      - name: Docker compose build ${{ matrix.app }}
+        run: |
+          LAST_TAG='$VERSION'
+          LAST_COMMIT='${{ github.sha }}'
+          docker-compose build ${{ matrix.app }}
+      - name: Check images
+        run: |
+          docker image ls
+      - name: Log in to registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
+
+      - id: tag-and-push
+        name: Tag and push (all branches)
+        run: |
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:${{ github.sha }}
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:$VERSION
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:${{needs.setup-env-vars.outputs.BRANCH_NAME}}
+
+      - id: tag-latest
+        name: Tag Latest (if main)
+        if: needs.setup-env-vars.outputs.BRANCH_NAME == 'main'
+        run: |
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:latest
+      - id: push-all-tags
+        name: Push All Tags
+        run: |
+          docker image ls
```

Comment:
Was this added for debugging purposes? Not sure why we'd need to list images at this point?